Cannot `make dist` on `--with-csc=mcs` tree

Given Roslyn is a pain for distributions to include, and that we can build the source tree with mcs, it seems like an oversight that we can't `make dist` after building the source tree with mcs.

```
make all-local
make[7]: Leaving directory '/home/directhex/Projects/mono/mcs'
make[6]: Leaving directory '/home/directhex/Projects/mono/mcs'
cp -p lib/build-darwin/Facades/System.Collections.Concurrent.dll lib/monolite-darwin/1051300002/Facades/System.Collections.Concurrent.dll
cp: cannot stat 'lib/build-darwin/Facades/System.Collections.Concurrent.dll': No such file or directory
Makefile:447: recipe for target 'lib/monolite-darwin/1051300002/Facades/System.Collections.Concurrent.dll' failed
make[5]: *** [lib/monolite-darwin/1051300002/Facades/System.Collections.Concurrent.dll] Error 1
make[5]: Leaving directory '/home/directhex/Projects/mono/mcs/class'
Makefile:459: recipe for target 'dist-monolite-all-platforms' failed
make[4]: *** [dist-monolite-all-platforms] Error 2
make[4]: Leaving directory '/home/directhex/Projects/mono/mcs/class'
build/rules.make:286: recipe for target 'dist-recursive' failed
make[3]: *** [dist-recursive] Error 1
make[3]: Leaving directory '/home/directhex/Projects/mono/mcs'
Makefile:985: recipe for target 'dist-hook' failed
make[2]: *** [dist-hook] Error 2
make[2]: Leaving directory '/home/directhex/Projects/mono'
Makefile:659: recipe for target 'distdir' failed
make[1]: *** [distdir] Error 2
make[1]: Leaving directory '/home/directhex/Projects/mono'
Makefile:760: recipe for target 'dist' failed
make: *** [dist] Error 2
```

--- a/mcs/class/Makefile
+++ b/mcs/class/Makefile
@@ -48,8 +48,8 @@ build_SUBDIRS :=  \
 	../tools/cil-stringreplacer
 
 ifdef MCS_MODE
-basic_PARALLEL_SUBDIRS := ../mcs
-build_PARALLEL_SUBDIRS := ../mcs
+basic_PARALLEL_SUBDIRS := ../mcs $(pcl_facade_dirs)
+build_PARALLEL_SUBDIRS := ../mcs $(pcl_facade_dirs)
 else
 basic_PARALLEL_SUBDIRS := $(pcl_facade_dirs)
 build_PARALLEL_SUBDIRS := $(pcl_facade_dirs)
@@ -423,7 +423,7 @@ all-local-aot:
 monolite_dir = monolite-$(HOST_PLATFORM)/$(MONO_CORLIB_VERSION)
 
 # Files needed to bootstrap C# compiler
-build_files = mscorlib.dll System.dll System.Xml.dll Mono.Security.dll System.Core.dll System.Security.dll System.Configuration.dll \
+build_files := mscorlib.dll System.dll System.Xml.dll Mono.Security.dll System.Core.dll System.Security.dll System.Configuration.dll \
 	System.Numerics.dll System.Xml.Linq.dll \
 	Facades/System.Collections.Concurrent.dll Facades/System.Linq.dll Facades/System.Runtime.dll Facades/System.Collections.dll	\
 	Facades/System.Reflection.Extensions.dll Facades/System.Text.Encoding.Extensions.dll Facades/System.Diagnostics.Debug.dll \

